.ci/install-r-deps.R

```diff
@@ -0,0 +1,76 @@
+args <- commandArgs(trailingOnly = TRUE)
+
+# which dependencies to install
+ALL_DEPS     <- "--all" %in% args
+BUILD_DEPS   <- ALL_DEPS || ("--build" %in% args)
+ROXYGEN_DEPS <- ALL_DEPS || ("--roxygen" %in% args)
+TEST_DEPS    <- ALL_DEPS || ("--test" %in% args)
+
+# force downloading of binary packages on macOS
+COMPILE_FROM_SOURCE <- "both"
+PACKAGE_TYPE <- getOption("pkgType")
+# CRAN has precompiled binaries for macOS and Windows... prefer those,
+# for faster installation.
+if (Sys.info()[["sysname"]] == "Darwin" || .Platform$OS.type == "windows") {
+    COMPILE_FROM_SOURCE <- "never"
+    PACKAGE_TYPE <- "binary"
+}
+options(
+    install.packages.check.source = "no"
+    , install.packages.compile.from.source = COMPILE_FROM_SOURCE
+)
+
+# always use the same CRAN mirror
+CRAN_MIRROR <- Sys.getenv("CRAN_MIRROR", unset = "https://cran.r-project.org")
+
+# we always want these
```

Comment:
Maybe we should allow installation of arbitrary R package by this script? For example, late `testthat` installation or `processx`.

Reply:
Yeah I agree! I implemented this in https://github.com/microsoft/LightGBM/commit/dae117f3d94d6c49873e6759b4da1b0a8bd7ea0e

Proposing adding two arguments, `--include` and `--exclude`, which are applied to the list of dependencies calculated *after* processing coarser arguments like `--all`, `--build`, and `--test`.